xilem/src/view/async_repeat.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{future::Future, marker::PhantomData, sync::Arc};
+
+use tokio::task::JoinHandle;
+use xilem_core::{DynMessage, Message, MessageProxy, NoElement, View, ViewId, ViewPathTracker};
+
+use crate::ViewCtx;
+
+pub fn async_repeat<M, F, H, State, Action, Fut>(
+    future_future: F,
+    on_event: H,
+) -> AsyncRepeat<F, H, M>
+where
+    F: Fn(MessageProxy<M>) -> Fut,
+    Fut: Future<Output = ()> + Send + 'static,
+    H: Fn(&mut State, M) -> Action + 'static,
+    M: Message + 'static,
+{
+    AsyncRepeat {
+        future_future,
+        on_event,
+        message: PhantomData,
+    }
+}
+
+pub struct AsyncRepeat<F, H, M> {
+    future_future: F,
+    on_event: H,
+    message: PhantomData<fn() -> M>,
+}
+
+impl<State, Action, F, H, M, Fut> View<State, Action, ViewCtx> for AsyncRepeat<F, H, M>
+where
+    F: Fn(MessageProxy<M>) -> Fut + 'static,
+    Fut: Future<Output = ()> + Send + 'static,
+    H: Fn(&mut State, M) -> Action + 'static,
+    M: Message + 'static,
+{
+    type Element = NoElement;
+
+    type ViewState = JoinHandle<()>;
+
+    fn build(&self, ctx: &mut ViewCtx) -> (Self::Element, Self::ViewState) {
+        let path: Arc<[ViewId]> = ctx.view_path().into();
+
+        let proxy = ctx.proxy.clone();
+        let handle = ctx
+            .runtime()
+            .spawn((self.future_future)(MessageProxy::new(proxy, path)));
+        // TODO: Clearly this shouldn't be a label here
```

Comment:
I don't understand this, do you mean `NoElement`?

Reply:
This comment is vestigal from before I added NoElement. Thanks for catching